src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfilesDebugLaunchProvider.cs

```diff
@@ -54,19 +54,37 @@ public override Task<bool> CanLaunchAsync(DebugLaunchOptions launchOptions)
         }
 
         /// <summary>
-        /// This is called to query the list of debug targets  
+        /// Called by StartupProjectRegistrar to determine whether this project should appear in the Startup list.
         /// </summary>
-        public override Task<IReadOnlyList<IDebugLaunchSettings>> QueryDebugTargetsAsync(DebugLaunchOptions launchOptions)
+        public async Task<bool> CanBeStartupProjectAsync(DebugLaunchOptions launchOptions)
         {
-            return QueryDebugTargetsInternalAsync(launchOptions, fromDebugLaunch: false);
+            try
+            {
+                ILaunchProfile activeProfile = await GetActiveProfileAsync();
+
+                // Now find the DebugTargets provider for this profile
+                IDebugProfileLaunchTargetsProvider? launchProvider = GetLaunchTargetsProvider(activeProfile);
+                if (launchProvider is null)
+                {
+                    return true;
+                }
```

Comment:
This statement can be removed. If `launchProvider` is null, it won't pass the next check and the lower `return true` will fire.